# Assignement 4
## Student Information
### Name: Tanzim Nawaz
### ID: 11834685

## Problem Statement
The task, based on my student ID, is to to perform T-Test to determine if there is a statistically significant difference in a dataset. 

I will be applying the T-Test on the tips dataset of seaborn python library. What I will be looking into is whether the tip amount is statistically significantly different between Lunch and Dinner time.

## Input Data Description 

I have opted to use the tips dataset from the seaborn library for this assignment to ensure originality and avoid common dataset choices. This is a real-world dataset that provides authentic restaurant service data, making it ideal for business analytics and statistical analysis.
The tips dataset contains 244 restaurant transactions collected from a single restaurant and includes the following features:

- tip: Tip amount in dollars (continuous variable, our primary variable of interest)
- total_bill: Total bill amount in dollars (continuous variable)
- sex: Customer gender (categorical: Male, Female)
- smoker: Smoking status (categorical: Yes, No)
- day: Day of the week (categorical: Thur, Fri, Sat, Sun)
- time: Meal period (categorical: Lunch, Dinner - our grouping variable)
- size: Party size/number of people (discrete variable: 1-6)

## Methodology

The very first step is to load the tips dataset from seaborn library.

Then I will show some basic information about the dataset.

Next, I will execute the T-Test with researchpy

Finally, I will write the result interpretation.

In [14]:
import pandas as pd
import seaborn as sns
import scipy.stats as stats
import researchpy as rp
import warnings

warnings.filterwarnings('ignore', category=FutureWarning)


The following section loads the dataset and displays some basic information.

In [ ]:
restaurant_tips_data = sns.load_dataset('tips')

In [ ]:
restaurant_tips_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   total_bill  244 non-null    float64 
 1   tip         244 non-null    float64 
 2   sex         244 non-null    category
 3   smoker      244 non-null    category
 4   day         244 non-null    category
 5   time        244 non-null    category
 6   size        244 non-null    int64   
dtypes: category(4), float64(2), int64(1)
memory usage: 7.4 KB


In [17]:
restaurant_tips_data.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [23]:
print("Dataset Shape:", restaurant_tips_data.shape)
print("Time categories:", restaurant_tips_data['time'].value_counts())

Dataset Shape: (244, 7)
Time categories: time
Dinner    176
Lunch      68
Name: count, dtype: int64


In [ ]:
# Separate data into groups for analysis
lunch_tips = restaurant_tips_data[restaurant_tips_data['time'] == 'Lunch']['tip']
dinner_tips = restaurant_tips_data[restaurant_tips_data['time'] == 'Dinner']['tip']

In [20]:
print(f"Lunch tips sample size: {len(lunch_tips)}")
print(f"Dinner tips sample size: {len(dinner_tips)}")

Lunch tips sample size: 68
Dinner tips sample size: 176


In [21]:
# Perform independent samples t-test using researchpy
test_summary, test_results = rp.ttest(
    group1=lunch_tips, 
    group1_name="Lunch",
    group2=dinner_tips, 
    group2_name="Dinner"
)

In [22]:
# Display results
print("\nT-Test Summary Statistics:")
display(test_summary)
print("\nT-Test Results:")
display(test_results)


T-Test Summary Statistics:


Variable      N      Mean        SD        SE  95% Conf.  Interval
0     Lunch   68.0  2.728088  1.205345  0.146170   2.436332  3.019844
1    Dinner  176.0  3.102670  1.436243  0.108261   2.889005  3.316335
2  combined  244.0  2.998279  1.383638  0.088578   2.823799  3.172758


T-Test Results:


Independent t-test   results
0  Difference (Lunch - Dinner) =    -0.3746
1           Degrees of freedom =   242.0000
2                            t =    -1.9063
3        Two side test p value =     0.0578
4       Difference < 0 p value =     0.0289
5       Difference > 0 p value =     0.9711
6                    Cohen's d =    -0.2722
7                    Hedge's g =    -0.2713
8               Glass's delta1 =    -0.3108
9             Point-Biserial r =    -0.1216

## Output and Result Interpretation

Looking at the T-Test result outputs, we can see that the two side test p value is 0.0578 which is greater that 0.05. This means that there is no statistically significant difference in the tip amount between Lunch and Dinner time.